src/core/ai/session/session-manager.test.ts

```
@@ -0,0 +1,741 @@
+        });
+
+        test('should handle periodic cleanup errors gracefully', async () => {
+            const consoleSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
+            const setIntervalSpy = vi.spyOn(global, 'setInterval');
+            mockStorageManager.database.list.mockRejectedValue(new Error('Storage error'));
+
+            await sessionManager.init();
+
+            // Access the interval function and call it directly to test error handling
+            const setIntervalCall = setIntervalSpy.mock.calls[0];
+            const cleanupFunction = setIntervalCall[0] as Function;
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)